### PR TITLE
commit oriyaf1

### PR DESCRIPTION
**Pull request by oriyaf1**

### CODE REVIEW

File: BH_Program/Analyzers/VideoAnalayzers/OcrNumberDetect.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        return True
+
+
+class NumberOCR:
+
+    def __init__(self, type='type0'):
+        self.type = type
+        self.npaClassifications = np.loadtxt("OcrFile/{0}classifications.txt".format(type),
+                                             np.float32)  # read in training classifications
+        self.npaFlattenedImages = np.loadtxt("OcrFile/" + type + "flattened_images.txt",
```
Comment (mbjahnoon) [inline BH_Program/Analyzers/VideoAnalayzers/OcrNumberDetect.py]: use string format like thos
np.loadtxt("OcrFile/{0}classifications.txt".format(type)

File: BH_Program/Analyzers/VideoAnalayzers/OcrNumberDetect.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                                             np.float32)  # read in training images
+        npaClassifications = self.npaClassifications.reshape(
+            (self.npaClassifications.size, 1))  # reshape numpy array to 1d, necessary to pass to call to train
+        self.kNearest = cv2.ml.KNearest_create()  # instantiate KNN object
+        self.kNearest.train(self.npaFlattenedImages, cv2.ml.ROW_SAMPLE, npaClassifications)
```
Comment (mbjahnoon) [inline BH_Program/Analyzers/VideoAnalayzers/OcrNumberDetect.py]: Mooved To constructor..
Good job

File: BH_Program/Analyzers/VideoAnalayzers/OcrFile/type0classifications.txt [hunk 3]
```diff
@@ -0,0 +1,94 @@
+5.200000000000000000e+01
```
[the remark this answers is not in the document]
Comment (oriyaf1) [reply BH_Program/Analyzers/VideoAnalayzers/OcrFile/type0classifications.txt]:
```suggestion
5.200000000000000000e+01
```
it not autoGenerate,
thr class GendData create it  : [https://github.com/mbjahnoon/BH-Highlights/blob/master/Python/GenData.py](url)

File: BH_Program/Analyzers/VideoAnalayzers/OcrNumberDetect.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
+# BH_Program/Analyzers/VideoAnalayzers/OcrNumberDetect.py
+
+import cv2
+import numpy as np
+import operator
+import os
+
+MIN_CONTOUR_AREA = 10
+RESIZED_IMAGE_WIDTH = 20
+RESIZED_IMAGE_HEIGHT = 30
```
[the remark this answers is not in the document]
Comment (oriyaf1) [reply BH_Program/Analyzers/VideoAnalayzers/OcrNumberDetect.py]: constructor its better, with adjustment to the scoreBoxDetect you writed.
i'ts good for my english!

File: BH_Program/Analyzers/VideoAnalayzers/OcrNumberDetect.py (hunk 1 above)
Comment (oriyaf1) [reply BH_Program/Analyzers/VideoAnalayzers/OcrNumberDetect.py]: ok

**Pull request merged**